### PR TITLE
[49] [UI] As a user, I can see the product color selection section on the Product Detail screen

### LINKED ISSUE #49: [UI] As a user, I can see the product color selection section on the Product Detail screen.
## Why

As a user, I can see the product color selection section on the Product Detail screen.
 
## Acceptance Criteria

The product color selection section on the Product Detail screen contains:

- "Color" title label on the top
- A color collection, each item contains:
  - A circle of which the background color performs the selected color for the product
  - A label showing the color name
  - Gray border for the item in the normal state
  - Blue border for the item in the selected state
- User can select the color

## Resource

image

### PR DESCRIPTION
**Pull request by nmint8m**
https://github.com/nimblehq/nimble-ecommerce-ios/issues/49

## What happened 👀

The product color selection section on the Product Detail screen contains:

- "Color" title label on the top
- A color collection, each item contains:
  - A circle of which the background color performs the selected color for the product
  - A label showing the color name
  - Gray border for the item in the normal state
  - Blue border for the item in the selected state
- User can select the color

## Insight

- Create `ColorCell` to perform a selection for color
- Create `ColorSelectionView` to perform a collection of colors
- Implement the selecting action for ColorSelectionView

### CODE REVIEW

File: ecommerce-ios/Views/ColorSelectionView/ColorSelectionView.swift [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    private static let horizontalSpacing: CGFloat = 34.0
+
+    private static let itemWidth: CGFloat = {
+        let numberOfColumns = CGFloat(Self.numberOfColumns)
+        return ((screenWidth - Self.spacing * (numberOfColumns + 1) - Self.horizontalSpacing) / numberOfColumns).rounded(.down)
+    }()
+
+    private static let columns: [GridItem] = {
+        [.init(.adaptive(minimum: itemWidth))]
+    }()
```
Comment (blyscuit) [inline ecommerce-ios/Views/ColorSelectionView/ColorSelectionView.swift]: again we don't need itemWidth
```
private static let columns: [GridItem] = {
        [
            .init(.flexible()),
            .init(.flexible())
        ]
    }()
```

Comment (nmint8m) [reply ecommerce-ios/Views/ColorSelectionView/ColorSelectionView.swift]: 💯 💯 💯 💯 💯

File: ecommerce-ios/Extensions/Color+Application.swift [hunk 2]
```diff
@@ -47,6 +47,9 @@ extension Color {
     /// #000000 Black with alpha 0.2
     static var blackAlpha2: Color { Color(hex: 0x000000, alpha: 0.2) }
 
+    /// #000000 Black with alpha 0.12
+    static var blackAlpha12: Color { Color(hex: 0x000000, alpha: 0.12) }
```
Comment (blyscuit) [inline ecommerce-ios/Extensions/Color+Application.swift]: somehow `blackAlpha2` is darker than `blackAlpha12`...

**Review by blyscuit — APPROVED** [review]

File: ecommerce-ios/Views/ColorSelectionView/ColorSelectionView.swift [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+                            }
+                    }
+                }
+            }
+        }
+    }
+
+    init(cellViewModels: [ColorCellViewModel]) {
+        self.cellViewModels = cellViewModels
+        selectedViewModel = cellViewModels.first!
```
Comment (vnntsu) [inline ecommerce-ios/Views/ColorSelectionView/ColorSelectionView.swift]: Fix potential crash here, it's merged then

**Review by vnntsu — APPROVED** [review]

**Pull request merged**